presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -198,6 +204,16 @@ public static Type typeConvert(Join.Type joinType)
         }
     }
 
+    public static boolean mustReplicateBuildSide(JoinNode node, Lookup lookup)
+    {
+        return isScalar(lookup.resolve(node.getRight())) || node.isCrossJoin();
+    }
+
+    public static boolean mustRepartition(JoinNode node)
```

Comment:
Having these methods in JoinNode will eventually cause the code to be extraced again for SemiJoinNode when we include that in reordering. We already have duplicated Enums